src/main/java/crawlercommons/filters/basic/BasicURLNormalizer.java

```diff
@@ -244,23 +261,25 @@ public String filter(String urlString) {
             // check for unnecessary use of "/../", "/./", and "//"
             try {
                 if (changed) {
-                    url = new URL(protocol, host, port, file);
+                    String tempUrl = protocol + "://" + (host == null ? "" : host) + (port == -1 ? "" : ":" + port) + file;
```

Comment:
This seems overtly complex. We could just return the concatenated string. Except we want explicitly verify that the normalized URL is both a valid URI and URL.

Reply:
... or use the URI internally for now. In `getFileWithNormalizedPath(url)` we convert again from URL to URI.